stdlib/builtins.pyi

```diff
@@ -1868,6 +1869,7 @@ class BaseException:
     __suppress_context__: bool
     __traceback__: TracebackType | None
     def __init__(self, *args: object) -> None: ...
+    def __new__(cls, *args: object, **kwds: Unused) -> Self: ...
```

Comment:
```suggestion
    def __new__(cls, *args: Any, **kwds: Any) -> Self: ...
```
Pyright is enforcing this section of the spec: https://typing.readthedocs.io/en/latest/spec/constructors.html#consistency-of-new-and-init. The signatures of `__new__` and `__init__` need to be consistent, i.e. they need to be able to materialize to the same static type. Therefore, we need to use an Any/Any signature, which materializes to any signature.

Pyright's error message I think has room for improvement here, so I'll report a bug about that.